src/imitation/scripts/train_preference_comparisons.py

```diff
@@ -155,22 +157,43 @@ def train_preference_comparisons(
             relabel_reward_fn=relabel_reward_fn,
         )
 
-    if trajectory_path is None:
+    if num_agents < 1:
+        raise ValueError("num_agents must be at least 1!")
+
+    def make_agent_trainer(seed: Optional[int] = None):
+        if agent_path is None:
+            agent = rl_common.make_rl_algo(venv)
+        else:
+            agent = rl_common.load_rl_algo_from_path(agent_path=agent_path, venv=venv)
+
         # Setting the logger here is not really necessary (PreferenceComparisons
         # takes care of that automatically) but it avoids creating unnecessary loggers
-        trajectory_generator = preference_comparisons.AgentTrainer(
+        return preference_comparisons.AgentTrainer(
             algorithm=agent,
             reward_fn=reward_net,
             venv=venv,
             exploration_frac=exploration_frac,
-            seed=_seed,
+            seed=_seed if seed is None else seed,
             custom_logger=custom_logger,
             **trajectory_generator_kwargs,
         )
+
+    if trajectory_path is None and num_agents == 1:
+        trajectory_generator = make_agent_trainer()
         # Stable Baselines will automatically occupy GPU 0 if it is available. Let's use
         # the same device as the SB3 agent for the reward model.
         reward_net = reward_net.to(trajectory_generator.algorithm.device)
+        allow_save_policy = True
+    elif trajectory_path is None and num_agents > 1:
+        members = [make_agent_trainer(_seed * i) for i in range(num_agents)]
```

Comment:
I suppose this is a way to generate pseudo-iid copies of the agent trainer. Might be a silly observation, but what if the seed is zero? Unlikely that someone would set that, but if they did they would start getting very bizarre errors and results without an apparent cause.

Reply:
Good point made it additive instead.